server/src/main/java/com/datastrato/gravitino/server/web/rest/TopicOperations.java

```diff
@@ -0,0 +1,204 @@
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/metalakes/{metalake}/catalogs/{catalog}/schemas/{schema}/topics")
+public class TopicOperations {
+  private static final Logger LOG = LoggerFactory.getLogger(TopicOperations.class);
+
+  private final TopicOperationDispatcher dispatcher;
+
+  @Context private HttpServletRequest httpRequest;
+
+  @Inject
+  public TopicOperations(TopicOperationDispatcher dispatcher) {
+    this.dispatcher = dispatcher;
+  }
+
+  @GET
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "list-topic." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  public Response listTopics(
+      @PathParam("metalake") String metalake,
+      @PathParam("catalog") String catalog,
+      @PathParam("schema") String schema) {
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            LOG.info("Listing topics under schema: {}.{}.{}", metalake, catalog, schema);
+            Namespace topicNS = Namespace.ofTopic(metalake, catalog, schema);
+            NameIdentifier[] topics =
+                TreeLockUtils.doWithTreeLock(
+                    NameIdentifier.of(metalake, catalog, schema),
+                    LockType.WRITE,
+                    () -> dispatcher.listTopics(topicNS));
+            return Utils.ok(new EntityListResponse(topics));
+          });
+    } catch (Exception e) {
+      return ExceptionHandlers.handleFilesetException(OperationType.LIST, "", schema, e);
+    }
+  }
+
+  @POST
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "create-topic." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "create-topic", absolute = true)
+  public Response createTopic(
+      @PathParam("metalake") String metalake,
+      @PathParam("catalog") String catalog,
+      @PathParam("schema") String schema,
+      TopicCreateRequest request) {
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            LOG.info(
+                "Creating topic under schema: {}.{}.{}.{}",
+                metalake,
+                catalog,
+                schema,
+                request.getName());
+            request.validate();
+            NameIdentifier ident =
+                NameIdentifier.ofTopic(metalake, catalog, schema, request.getName());
+
+            Topic topic =
+                TreeLockUtils.doWithTreeLock(
+                    ident,
+                    LockType.WRITE,
+                    () ->
+                        dispatcher.createTopic(
+                            ident, request.getComment(), null, request.getProperties()));
```

Comment:
Can you please add a comment to explain why it is null here?

Reply:
added